### ISSUE TITLE
NO LISTEN PORT

### ISSUE THREAD

**Issue by Ikki-Dai** [ClickHouse/ClickHouse#35235]
I have uncommit below in config.xml and restart service
```xml
<listen_host>0.0.0.0</listen_host>
```
but not foud any avaliable PORT in LISTEN status
```bash
 root@VM-4-7-centos  ~  netstat -tulpn | grep LISTEN
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      1498/master
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      1393/nginx: master
tcp        0      0 0.0.0.0:6379            0.0.0.0:*               LISTEN      1352/redis-server 0
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      1393/nginx: master
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1344/sshd
```

so while try to connect caused below
```bash
root@VM-4-7-centos  ~ clickhouse-client
ClickHouse client version 22.2.2.1.
Connecting to localhost:9000 as user default.
Code: 210. DB::NetException: Connection refused (localhost:9000). (NETWORK_ERROR)

root@VM-4-7-centos  ~  clickhouse-client -h 127.0.0.1
ClickHouse client version 22.2.2.1.
Connecting to 127.0.0.1:9000 as user default.
Code: 210. DB::NetException: Connection refused (127.0.0.1:9000). (NETWORK_ERROR)
```

**Comment by den-crane** [comment]
check that CH server is running
```
ps -ef|grep clickhouse-server
```

**Comment by Ikki-Dai** [comment]
yes, clickhouse-server is running.
i not sure whether is related to openssl keys.
i generate server.key and server.crt under /etc/clickhouse-server/

**Comment by genzgd** [comment]
Does your server have an IPv6 address?  If you only enable `<listen_host>0.0.0.0</listen_host>` ClickHouse does not listen on IPv6.  Does clickhouse-client 127.0.0.1 work?  If that does work does your server log (probably at `/var/lib/clickhouse-server/clickhouse-server.log` show any errors on startup?

**Comment by Ikki-Dai** [comment]
yes, server have IPv6 address, and `clickhouse-client -h 127.0.0.1` doesn't work

**Comment by genzgd** [comment]
Also of course do you have `<tcp_port>9000</tcp_port>` in your config?

**Comment by Ikki-Dai** [comment]
sure

**Comment by genzgd** [comment]
Well you should see lines like this in your log file:
```
2022.03.13 20:13:53.208112 [ 8636538 ] {} <Information> Application: Listening for http://0.0.0.0:8123
2022.03.13 20:13:53.208123 [ 8636538 ] {} <Information> Application: Listening for native protocol (tcp): 0.0.0.0:9000
2022.03.13 20:13:53.208134 [ 8636538 ] {} <Information> Application: Listening for replica communication (interserver): http://0.0.0.0:9009
2022.03.13 20:13:53.208144 [ 8636538 ] {} <Information> Application: Listening for MySQL compatibility protocol: 0.0.0.0:9004
2022.03.13 20:13:53.208152 [ 8636538 ] {} <Information> Application: Listening for PostgreSQL compatibility protocol: 0.0.0.0:9008
2022.03.13 20:13:53.208155 [ 8636538 ] {} <Information> Application: Ready for connections.
2022.03.13 20:13:53.208574 [ 8636676 ] {} <Debug> DNSResolver: Updated DNS cache
```

**Comment by Ikki-Dai** [comment]
seems not found after restart.

**Comment by Ikki-Dai** [comment]
```
2022.03.11 22:30:10.019389 [ 9536 ] {} <Information> Application: Forked a child process to watch
2022.03.11 22:30:10.019668 [ 9536 ] {} <Information> SentryWriter: Sending crash reports is disabled
2022.03.11 22:30:10.019751 [ 9536 ] {} <Trace> Pipe: Pipe capacity is 1.00 MiB
2022.03.11 22:30:10.020603 [ 9523 ] {} <Information> Application: Will watch for the process with pid 9536
2022.03.11 22:30:10.414045 [ 9536 ] {} <Information> : Starting ClickHouse 22.2.2.1 with revision 54459, build id: 5F3D9E4F48D4CC47, PID 9536
2022.03.11 22:30:10.414234 [ 9536 ] {} <Information> Application: starting up
2022.03.11 22:30:10.414265 [ 9536 ] {} <Information> Application: OS name: Linux, version: 3.10.0-1160.59.1.el7.x86_64, architecture: x86_64
2022.03.11 22:30:10.471429 [ 9536 ] {} <Trace> AsynchronousMetrics: Scanning /sys/class/thermal
2022.03.11 22:30:10.471469 [ 9536 ] {} <Trace> AsynchronousMetrics: Scanning /sys/block
2022.03.11 22:30:10.471549 [ 9536 ] {} <Trace> AsynchronousMetrics: Scanning /sys/devices/system/edac
2022.03.11 22:30:10.471569 [ 9536 ] {} <Trace> AsynchronousMetrics: Scanning /sys/class/hwmon
2022.03.11 22:30:10.887134 [ 9536 ] {} <Warning> Application: Calculated checksum of the binary: 03A21B8EF25D04A4DCD8C0FCA8310FDA. There is no information about the reference checksum.
2022.03.11 22:30:10.887934 [ 9536 ] {} <Information> Application: It looks like the process has no CAP_IPC_LOCK capability, binary mlock will be disabled. It could happen due to incorrect ClickHouse package installation. You could resolve the problem manually with 'sudo setcap cap_ipc_lock=+ep /usr/bin/clickhouse'. Note that it will not work on 'nosuid' mounted filesystems.
2022.03.11 22:30:10.894470 [ 9536 ] {} <Information> StatusFile: Status file /var/lib/clickhouse/status already exists - unclean restart. Contents:
PID: 6229
Started at: 2022-03-11 22:19:19
Revision: 54459

2022.03.11 22:30:10.896081 [ 9536 ] {} <Debug> Application: rlimit on number of file descriptors is 500000
2022.03.11 22:30:10.896102 [ 9536 ] {} <Debug> Application: Initializing DateLUT.
2022.03.11 22:30:10.896107 [ 9536 ] {} <Trace> Application: Initialized DateLUT with time zone 'Asia/Shanghai'.
2022.03.11 22:30:10.896145 [ 9536 ] {} <Debug> Application: Setting up /var/lib/clickhouse/tmp/ to store temporary data in it
2022.03.11 22:30:10.898614 [ 9536 ] {} <Debug> Application: Configuration parameter 'interserver_http_host' doesn't exist or exists and empty. Will use 'VM-4-7-centos' as replica host.
2022.03.11 22:30:10.898643 [ 9536 ] {} <Debug> Application: Initiailizing interserver credentials.
2022.03.11 22:30:10.898781 [ 9536 ] {} <Information> SensitiveDataMaskerConfigRead: 1 query masking rules loaded.
2022.03.11 22:30:10.902263 [ 9536 ] {} <Debug> ConfigReloader: Loading config '/etc/clickhouse-server/config.xml'
2022.03.11 22:30:10.913758 [ 9536 ] {} <Debug> ConfigReloader: Loaded config '/etc/clickhouse-server/config.xml', performing update on configuration
2022.03.11 22:30:10.915298 [ 9536 ] {} <Information> Application: Setting max_server_memory_usage was set to 1.62 GiB (1.79 GiB available * 0.90 max_server_memory_usage_to_ram_ratio)
2022.03.11 22:30:10.924291 [ 9536 ] {} <Debug> CertificateReloader: Reloading certificate (/etc/clickhouse-server/server.crt) and key (/etc/clickhouse-server/server.key).
2022.03.11 22:30:10.925717 [ 9536 ] {} <Information> CertificateReloader: Reloaded certificate (/etc/clickhouse-server/server.crt) and key (/etc/clickhouse-server/server.key).
2022.03.11 22:30:10.925736 [ 9536 ] {} <Debug> CertificateReloader: Initializing certificate reloader.
2022.03.11 22:30:10.927638 [ 9536 ] {} <Debug> ConfigReloader: Loaded config '/etc/clickhouse-server/config.xml', performed update on configuration
2022.03.11 22:30:10.931767 [ 9536 ] {} <Debug> ConfigReloader: Loading config '/etc/clickhouse-server/users.xml'
2022.03.11 22:30:10.933861 [ 9536 ] {} <Debug> ConfigReloader: Loaded config '/etc/clickhouse-server/users.xml', performing update on configuration
2022.03.11 22:30:10.937337 [ 9536 ] {} <Debug> ConfigReloader: Loaded config '/etc/clickhouse-server/users.xml', performed update on configuration
2022.03.11 22:30:10.937887 [ 9536 ] {} <Debug> Access(user directories): Added users.xml access storage 'users.xml', path: /etc/clickhouse-server/users.xml
2022.03.11 22:30:10.943247 [ 9536 ] {} <Debug> Access(user directories): Added local directory access storage 'local directory', path: /var/lib/clickhouse/access/
2022.03.11 22:30:10.943299 [ 9536 ] {} <Information> Application: Uncompressed cache size was lowered to 918.93 MiB because the system has low amount of memory
2022.03.11 22:30:10.943696 [ 9536 ] {} <Information> Context: Initialized background executor for merges and mutations with num_threads=16, num_tasks=32
2022.03.11 22:30:10.943714 [ 9536 ] {} <Information> Context: Initialized background executor for move operations with num_threads=8, num_tasks=8
2022.03.11 22:30:10.943724 [ 9536 ] {} <Information> Context: Initialized background executor for fetches with num_threads=8, num_tasks=8
2022.03.11 22:30:10.943737 [ 9536 ] {} <Information> Context: Initialized background executor for common operations (e.g. clearing old parts) with num_threads=8, num_tasks=8
2022.03.11 22:30:10.943750 [ 9536 ] {} <Information> Application: Mark cache size was lowered to 918.93 MiB because the system has low amount of memory
2022.03.11 22:30:10.943802 [ 9536 ] {} <Information> Application: Loading user defined objects from /var/lib/clickhouse/
2022.03.11 22:30:10.943811 [ 9536 ] {} <Debug> UserDefinedSQLObjectsLoader: loading user defined objects
2022.03.11 22:30:10.944583 [ 9536 ] {} <Debug> Application: Loaded user defined objects
2022.03.11 22:30:10.944597 [ 9536 ] {} <Information> Application: Loading metadata from /var/lib/clickhouse/
2022.03.11 23:47:48.425401 [ 9537 ] {} <Trace> BaseDaemon: Received signal 15
2022.03.11 23:47:48.433822 [ 9537 ] {} <Information> Application: Received termination signal (Terminated)
```
